static/js/zamboni/validator.js

```diff
@@ -194,71 +213,128 @@ function initValidator($doc) {
         return this.tiers[tierId];
     };
 
+    MsgVisitor.prototype.filterMessage = function(msg) {
+        if (this.hideIgnored && msg.ignored)
+            return false;
+
+        if (this.hideNonSigning && !msg.signing_severity)
```

Comment:
What did `else` ever do to you? You could drop the `if`s and just return the boolean expression also.

```js
return !(this.hideIgnored && msg.ignored)
       && !(this.hideNonSigning && !msg.signing_severity);